File: src/main/java/um/haberes/rest/model/FacultadCarreraMateria.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinColumns;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "facultadId", "planId", "carreraId", "materiaId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FacultadCarreraMateria extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 9020765877839550829L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long facultadCarreraMateriaId;

	private Integer facultadId;
	private Integer planId;
	private Integer carreraId;
	private String materiaId;

	@OneToOne
	@JoinColumns({
			@JoinColumn(name = "facultadId", referencedColumnName = "facultadId", insertable = false, updatable = false),
			@JoinColumn(name = "planId", referencedColumnName = "planId", insertable = false, updatable = false),
			@JoinColumn(name = "carreraId", referencedColumnName = "carreraId", insertable = false, updatable = false) })
	private FacultadCarrera facultadCarrera;

	@OneToOne
	@JoinColumns({
			@JoinColumn(name = "facultadId", referencedColumnName = "facultadId", insertable = false, updatable = false),
			@JoinColumn(name = "planId", referencedColumnName = "planId", insertable = false, updatable = false),
			@JoinColumn(name = "materiaId", referencedColumnName = "materiaId", insertable = false, updatable = false) })
	private FacultadMateria facultadMateria;

}

File: src/main/java/um/haberes/rest/kotlin/model/FacultadCarreraMateria.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["facultadId", "planId", "carreraId", "materiaId"])])
data class FacultadCarreraMateria(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var facultadCarreraMateriaId: Long? = null,

    var facultadId: Int? = null,
    var planId: Int? = null,
    var carreraId: Int? = null,
    var materiaId: String? = null,

    @OneToOne
    @JoinColumns(
        JoinColumn(
            name = "facultadId",
            referencedColumnName = "facultadId",
            insertable = false,
            updatable = false
        ),
        JoinColumn(name = "planId", referencedColumnName = "planId", insertable = false, updatable = false),
        JoinColumn(name = "carreraId", referencedColumnName = "carreraId", insertable = false, updatable = false)
    )
    var facultadCarrera: FacultadCarrera? = null,

    @OneToOne
    @JoinColumns(
        JoinColumn(
            name = "facultadId",
            referencedColumnName = "facultadId",
            insertable = false,
            updatable = false
        ),
        JoinColumn(name = "planId", referencedColumnName = "planId", insertable = false, updatable = false),
        JoinColumn(name = "materiaId", referencedColumnName = "materiaId", insertable = false, updatable = false)
    )
    var facultadMateria: FacultadMateria? = null

) : Auditable()
